### DISCUSSION TITLE
Error loading script: tag_autocomplete_helper.py

### DISCUSSION THREAD

**Discussion by StallmanSlave** [DominikDoom/a1111-sd-webui-tagcomplete#73] [Q&A] [answered]
Having trouble with this, a bunch of runtime errors have popped up relating to this so I find it hard to keep track of progress in rendering images regardless of if this failed to work or not. Don't know how to remove it or fix it to make it operable. 
```
Error loading script: tag_autocomplete_helper.py
Traceback (most recent call last):
  File "D:\Torrents\StableDiffusion\Working directory\stable-diffusion-webui-master\modules\scripts.py", line 71, in load_scripts
    exec(compiled, module.__dict__)
  File "D:\Torrents\StableDiffusion\Working directory\stable-diffusion-webui-master\scripts\tag_autocomplete_helper.py", line 14, in <module>
    TAGS_PATH = Path(scripts.basedir()).joinpath('tags')
AttributeError: module 'modules.scripts' has no attribute 'basedir'
```

**Comment by StallmanSlave** [comment]
I went through each file and deleted it, I think it was because I did not replace the .gitignore that was already in the root directory when I tried that method. The extensions method doesn't seem to do anything. I'm guessing the gitignore is needed to pull stuff or something.

**Comment by DominikDoom** [comment, marked as answer]
scripts.basedir is a function needed for this to run when installed as an extension, if it's not found that means your webui is/was outdated.
Gitignore should have no relation to this, that just tells Git what not to include when tracking changes.
It can prevent pulling if you have edited it in some way before, but would state that in the pull message.

**Reply by StallmanSlave** [reply]
So, what's the course of action? I would give you the version of the SD I'm using but I don't know whare it is, it just says main so I assume it was recent, I DL'd it early October. I'm using windows so I'm not sure how to actually git pull and update the thing to use the new extensions feature. I apologize for the inadequacy on my part to provide more useful information on resolving the problem (I don't have miniconda insalled, or the batch file auto DL'ed it, idk, only see python and git in parent directory above SD. I do have powershell and command prompt)

**Reply by DominikDoom** [reply]
For the webui itself: If you installed it using git clone (instead of just downloading and unzipping the folder), you should be able to just open a powershell or cmd window in the webui's installation root (where your webui-user.bat etc. are) and type `git pull` as a command. If anything prevents it, it will tell you.

For this script: pretty much the same, see https://github.com/DominikDoom/a1111-sd-webui-tagcomplete#as-an-extension-recommended for a bit more detail.

**Reply by StallmanSlave** [reply]
ah, It looked like I downloaded it as zip so it's not a git repo. Would downloading it as zip again and extracting it to the working directory do the thing? it could replace any files I have edited so I think it's bad since I need the options in webui. Would just doing `git clone https://github.com/AUTOMATIC1111/stable-diffusion-webui.git` in the SD folder just add the .git and then read the present files and compare it to the master branch and DL the updated files while leaving the user altered files (and added files) alone?

**Reply by DominikDoom** [reply]
Downloading and extracting manually might work, but it has the negative that it doesn't remove old or renamed files which could lead to problems.

Just doing git clone won't work since it wants an empty directory, but you can use the following to set it up with existing files:
```bash
git init
git remote add origin https://github.com/AUTOMATIC1111/stable-diffusion-webui.git
git fetch
git reset origin/master
git checkout -t origin/master
```
That should create the local git repo and tell it to pull from the webui GitHub as origin. Should work without needing to reinstall the webui.

**Reply by StallmanSlave** [reply]
Thank you very much, I'm using brave (chromium), the auto tagging still doesn't work. I had to use `git merge master` instead of checkout -t because it said it already exists but it said the master is already updated. I don't really know if this is solved or not but it doesn't seem like I could do anything else and I don't think any further help will be useful past this point. I just need to monkey around, is there a special button to press for the tagging suggestions/completions to show up?

**Reply by StallmanSlave** [reply]
I'mlooking at https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Extensions Now I might know why it doesn't work, are there dependencies I'm ignorant of that is needed for this to work, I don't have other extensions at the moment.

**Reply by DominikDoom** [reply]
No, if you are on a recent webui version extensions are loaded by default. As long as all files the script needs are in place, autocompletion will just show with typing as soon as it's finished loading the tags, which shouldn't take more than a few seconds. If you clone the script into the extensions directory like suggested in the readme, you should have all you need.

**Reply by DominikDoom** [reply]
Do you have this extensions tab in your Webui? If yes your version is new enough, if not the commands I suggested might not have worked as intended. You should also be able to find this script inside the "Available" list in that tab and install it that way if you're not sure about placing it in the right spot.
![image](https://user-images.githubusercontent.com/34448969/201529102-465984d8-d142-4070-8dd1-3a6b82eeaed8.png)

**Reply by StallmanSlave** [reply]
Yeah, I did everything.. but it just doesn't work. Think I'm just going to try git cloning the automatics repo in separate directory and move all the configs and models piece by piece over from there.

**Reply by StallmanSlave** [reply]
> Do you have this extensions tab in your Webui?

No it didn't It must've not worked. gonna do what I said above and I'll let you know if it all worked then.
